### PR TITLE
doc: add OFFBOARDING.md and update ONBOARDING.md

### PR DESCRIPTION
**Pull request by AshCripps**
Ive added an OFFBOARDING.md which contains a checklist for offboarding
members. Ive also updated ONBOARDING.md to have similar checklists for
adding members and also removed some duplicate information

Let me know if you have any suggestions/ if ive missed anything from the
checklists.

Ive tried to write the document in a style so that it can just be copied pasted into
an issue for the on/offboarding of the member

### CODE REVIEW

File: OFFBOARDING.md [hunk 1]
```diff
@@ -0,0 +1,18 @@
+# Node.js build WG offboarding
+
+This document serves as a checklist of what we must go through when offboarding a member of the working group.
+
+
+### Offboarding issue checklist
+
+- [ ] Remove their GPG key from `test` group in `nodejs-private/secrets`
+- [ ] Remove their GPG key from `release` group in `nodejs-private/secrets` (if applicable)
+- [ ] Remove their GPG key from `infra` group in `nodejs-private/secrets` (if applicable)
```
Comment (sam-github) [inline OFFBOARDING.md]:
```suggestion
- [ ] Remove their GPG key from `infra` group in `nodejs-private/secrets` (if applicable)
- [ ] Remove their GPG key from `infra-macstadium` group in `nodejs-private/secrets` (if applicable)
- [ ] Remove their GPG key from `github-bot` group in `nodejs-private/secrets` (if applicable)
```

File: OFFBOARDING.md [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+- [ ] Remove them from build teams they are a member of:
+    - [ ] nodejs/build
+    - [ ] nodejs/build-release
+    - [ ] nodejs/build-infra
+    - [ ] nodejs/jenkins-admins
+    - [ ] nodejs/jenkins-release-admins
```
Comment (sam-github) [inline OFFBOARDING.md]:
```suggestion
    - [ ] nodejs/jenkins-release-admins
    - [ ] nodejs-private/build
```

File: ONBOARDING.md [hunk 3]
```diff
@@ -3,67 +3,41 @@
...
-* Members/"who does what"/interest areas
-* Technologies and software in use
+- [ ] Email the new member and obtain their public GPG key
+- [ ] Add them to the `test` group in `nodejs-private/secrets`
+- [ ] Add them to the `nodejs/build` and `nodejs-private/build` GitHub teams
+- [ ] Schedule a meeting with the member to:
+    - [ ] Walk them through the infrastructure and what other members do 
+    - [ ] Explain how to decrypt nodejs/secrets 
+    - [ ] Practice running the `jenkins/worker/create.yml` playbook
+  on one of the machines in the test CI cluster
```
Comment (sam-github) [inline ONBOARDING.md]: accidental hard line break?

File: ONBOARDING.md [hunk 4]
```diff
@@ -3,67 +3,41 @@
...
+    - [ ] Answer any questions they may have
+- [ ] Before the meeting the Onboardee is to:
```
Comment (sam-github) [inline ONBOARDING.md]: move the "before the meeting" to before the checklist of things to do in the meeting?

File: ONBOARDING.md [hunk 5]
```diff
@@ -3,67 +3,41 @@
...
+  - [ ] Install Ruby and the `dotgpg` gem
+  - [ ] Install Python and Ansible
+  - [ ] Read `services.md`. `jenkins-guide.md`,
+    `GOVERNANCE.md` and to try and have any questions ready
```
Comment (sam-github) [inline ONBOARDING.md]:
```suggestion
    `GOVERNANCE.md` and have any questions ready
```

File: OFFBOARDING.md (hunk 2 above)
Comment (richardlau) [inline OFFBOARDING.md]: add
```
    - [ ] nodejs-private/build
```

File: ONBOARDING.md [hunk 6]
```diff
@@ -3,67 +3,41 @@
...
 
-### CI/Jenkins
 
-* Public CI (ci.nodejs.org)
-  * `nodejs/jenkins-admins` are administrators
-  * Tests Node.js itself, and several other Foundation projects
-* Release CI (ci-release.nodejs.org)
-  * `nodejs/jenkins-release-admins` are administrators
-  * What creates and publishes all releases of Node.js
+All members that are onboarded to the build-release and build-infra teams
```
Comment (sam-github) [inline ONBOARDING.md]: I don't know if its useful to have these last couple. For one thing, the emailing is unnecessary, the key are already in nodejs/secrets/build/test (assuming that no one is added to the more critical release or infra if they aren't already in build, which is always true AFAIK). For another, there are lots of little bits of extra permissions that can be given, maybe best just list them all as a single checklist?

File: ONBOARDING.md [hunk 7]
```diff
@@ -3,67 +3,41 @@
...
-* IRC (specifically `#node-build` is important for communicating with
-  other Node.js project members, and how we receive many initial signals
-  of downtime
-    * IRC logs are maintained at http://logs.libuv.org/node-build
-* Build WG meetings are every 2-3 weeks, and you should try and attend
-  as many as possible. The meetings are listed in the [Node.js
-Foundation calendar][].
 
 [`nodejs-private/secrets` repository]: https://github.com/nodejs-private/secrets
 [Node.js Foundation calendar]: https://nodejs.org/calendar
```
Comment (sam-github) [inline ONBOARDING.md]: can be removed now its unrefed

**Comment by AshCripps** [comment]
Ive addressed comments and have updated with a list of "further permissions". PTAL and tell me if theres anything obvious missing.

File: ONBOARDING.md [hunk 8]
```diff
@@ -3,67 +3,34 @@
...
+- [ ] Schedule a meeting with the member to:
+    - [ ] Walk them through the infrastructure and what other members do
+    - [ ] Explain how to decrypt nodejs/secrets 
+    - [ ] Practice running the `jenkins/worker/create.yml` playbook on one of the machines in the test CI cluster
+    - [ ] Answer any questions they may have
+- [ ] PR changes to [README.md](./README.md#build-wg-members) to add the member to build-test
+
+
+At first all members are added to nodejs/build-test. As their contribution and experience grows
+within the working group they may be granted further permissions and access. Examples of further access include:
```
Comment (sam-github) [inline ONBOARDING.md]: I suggest removing from "Examples of.." right to the end. They are all examples, but this is better discussed in access.md, its not part of the the onboarding checklist. Actually, everything from line 26 above down (the stuff after the checklist) I think could just be deleted without being missed.

File: ONBOARDING.md [hunk 9]
```diff
@@ -3,67 +3,25 @@
...
+- [ ] Schedule a meeting with the member to:
+    - [ ] Walk them through the infrastructure and what other members do
+    - [ ] Explain how to decrypt nodejs/secrets 
+    - [ ] Practice running the `jenkins/worker/create.yml` playbook on one of the machines in the test CI cluster
+    - [ ] Answer any questions they may have
+- [ ] PR changes to [README.md](./README.md#build-wg-members) to add the member to build-test
+
+
+At first all members are added to nodejs/build-test. Members may be granted further permissions and access as needed
+by the working group
```
Comment (sam-github) [inline ONBOARDING.md]:
```suggestion
by the working group, see [GOVERNANCE][./GOVERNANCE.md].
```

**Comment by AshCripps** [comment]
@nodejs/build this should be ready now PTAL

File: OFFBOARDING.md [hunk 10]
```diff
@@ -0,0 +1,23 @@
...
+- [ ] Remove their GPG key from `infra-macstadium` group in `nodejs-private/secrets`
+- [ ] Remove their GPG key from `github-bot` group in `nodejs-private/secrets`
+- [ ] Remove them from build teams they are a member of:
+    - [ ] nodejs/build
+    - [ ] nodejs/build-release
+    - [ ] nodejs/build-infra
+    - [ ] nodejs/jenkins-admins
+    - [ ] nodejs/jenkins-release-admins
+    - [ ] nodejs-private/build
+- [ ] PR changes to [README.md](./README.md#build-wg-members) to move the member to emeritus status
```
Comment (richardlau) [inline OFFBOARDING.md]: We could also say to run `ncu team sync` which would validate the user has been removed from the build teams.

Comment (AshCripps) [reply OFFBOARDING.md]: added

**Pull request merged**